File: solutions/kotlin/rotational-cipher/src/main/kotlin/RotationalCipher.kt
class RotationalCipher(private val shiftKey: Int) {

    fun encode(text: String): String = text.map {
        when (it) {
            in 'a'..'z' -> rotate(it, 'a'.code)
            in 'A'..'Z' -> rotate(it, 'A'.code)
            else -> it
        }
    }.joinToString("")
    
    private fun rotate(c: Char, start: Int): Char =
       (start + (c.code + shiftKey - start) % 26).toChar()
}

File: solutions/go/rotational-cipher/rotational_cipher.go
// Package rotationalcipher implementats the rotational cipher,
// also sometimes called the Caesar cipher.
package rotationalcipher

// RotationalCipher returns a rotated string when given a plaintext and
// amount to shift by. Each letter is shifted for the value of the key.
func RotationalCipher(plain string, shiftKey int) string {
	output := []rune(plain)
	for i, char := range output {
		if char >= 'a' && char <= 'z' {
			output[i] = rotate(char, rune(shiftKey), 'a')
		} else if char >= 'A' && char <= 'Z' {
			output[i] = rotate(char, rune(shiftKey), 'A')
		}
	}
	return string(output)
}

func rotate(c, n, start rune) rune {
	return start + (c+n-start)%26
}
